### PR TITLE
alpine: Reduce size and startup time

### PR DESCRIPTION
**Pull request by qinqon**
The alpine qcow2 image can be reduced by reserving less space and removing packages also the startup time can be reduced by removing guest agent

### CODE REVIEW

File: cluster-provision/images/vm-image-builder/alpine-cloud-init/configure.sh
Comment (xpivarc) [inline cluster-provision/images/vm-image-builder/alpine-cloud-init/configure.sh:9]: Can you comment on this change?

File: cluster-provision/images/vm-image-builder/alpine-cloud-init/create-image.sh
Comment (ormergi) [inline cluster-provision/images/vm-image-builder/alpine-cloud-init/create-image.sh:15]: nit: you can pass `--workdir $(pwd)` instead of passing it to `sh -c `

Just to understand, why is it necessary to mount /lib/modules and /dev? is it safe? 
would it work if we mount it as read only?

File: cluster-provision/images/vm-image-builder/alpine-cloud-init/packages
Comment (ormergi) [inline cluster-provision/images/vm-image-builder/alpine-cloud-init/packages:8]: Is it necessary for kubevirt tests?

File: cluster-provision/images/vm-image-builder/alpine-cloud-init/create-image.sh
Comment (qinqon) [reply cluster-provision/images/vm-image-builder/alpine-cloud-init/create-image.sh:15]: --privileged is supose to be mounting those already don't know why I added them, I will remove and use --workdir.

File: cluster-provision/images/vm-image-builder/alpine-cloud-init/packages
Comment (qinqon) [reply cluster-provision/images/vm-image-builder/alpine-cloud-init/packages:8]: Not yet, let's go step by step, so we have the fastest alpline cloud-init image possible.

**Review by EdDev — COMMENTED** [review]
> The alpine qcow2 image can be reduced by reseving less space for the
disk.

- Typo with `reseving`.
- The removal of the guest-agent is reducing the usage potential significantly.
   Can we control this through cloud-init user-data? i.e. tell it to run only if we explicitly ask for it there.
   I agree that by default it should not be running.

**Review by dhiller — APPROVED** [review]
/approve

Have verified that the image works. Holding for remaining reviewers to settle. Please remove hold when it's ready and ping me again in case you need another lgtm.

/hold

**Comment by qinqon** [comment]
/hold cancel

Tested by @AlonaKaplan at https://github.com/kubevirt/kubevirt/pull/7520#issuecomment-1096072028

**Comment by qinqon** [comment]
> /approve
> 
> Have verified that the image works. Holding for remaining reviewers to settle. Please remove hold when it's ready and ping me again in case you need another lgtm.
> 
> /hold

@dhiller can you push an image to kubevirtci with this similar to one one we did already (it would be better to have some automation for this though).

**Comment by qinqon** [comment]
/override check-provision-k8s-1.23 check-provision-k8s-1.22-ipv6

**Pull request merged**
